client/landing/gutenboarding/hooks/use-site-title.ts

```diff
@@ -13,11 +13,14 @@ import usePodcastTitle from './use-podcast-title';
 export default function useSiteTitle(): void {
 	const { setSiteTitle } = useDispatch( ONBOARD_STORE );
 	const podcastTitle = usePodcastTitle();
+	const { hasSiteTitle } = useSelect( ( select ) => ( {
+		hasSiteTitle: select( ONBOARD_STORE ).getSelectedSiteTitle().length > 0,
```

Comment:
Wondering whether [getSelectedSiteTitle()](https://github.com/Automattic/wp-calypso/blob/9b97938fde2e532545458ff937891bd4185b483e/client/landing/gutenboarding/stores/onboard/selectors.ts#L15) could not be a string, in which case maybe we should take care about getting the length of an `undefined` value, for instance.

Reply:
This is where Typescript helps us out. I don't think the optional chaining operator (`?.`) is needed here, as the compiler reads that `getSelectedSiteTitle` as always returning a string.

<img width="809" alt="Screen Shot 2021-02-17 at 14 41 40" src="https://user-images.githubusercontent.com/1699996/108265482-72990f80-712e-11eb-8c40-a30a4c80da6c.png">

Reply:
Additionally it looks like [the reducer returns a string](https://github.com/Automattic/wp-calypso/blob/52a87ebd8707f9f8f349a2b8047f2f0eb910e7ab/client/landing/gutenboarding/stores/onboard/reducer.ts#L207), one way or the other.

Reply:
Nice thanks for clarifying! Always good to check. 😅